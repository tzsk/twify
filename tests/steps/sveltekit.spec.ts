import fs from 'fs-extra';
import {
  movePostCSS,
  setupLayoutFile,
} from '../../src/frameworks/steps/sveltekit';

describe('SvelteKit Steps', () => {
  beforeAll(() => {
    vi.stubGlobal('console', { ...console, log: vi.fn() });
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  afterAll(() => {
    vi.resetAllMocks();
  });

  it('can setup layout file if it does not exist in ts', async () => {
    vi.spyOn(fs, 'existsSync')
      .mockReturnValueOnce(true)
      .mockReturnValueOnce(false);
    vi.spyOn(fs, 'ensureFile').mockResolvedValue();
    const writeSpy = vi.spyOn(fs, 'writeFile').mockResolvedValue();

    await setupLayoutFile();
    const [fileName, content] = writeSpy.mock.lastCall || [];

    expect(fileName).toMatch('src/routes/+layout.svelte');
    expect(content).toMatchSnapshot();
  });

  it('can setup layout file if it does not exist in js', async () => {
    vi.spyOn(fs, 'existsSync')
      .mockReturnValueOnce(false)
      .mockReturnValueOnce(false);
    vi.spyOn(fs, 'ensureFile').mockResolvedValue();
    const writeSpy = vi.spyOn(fs, 'writeFile').mockResolvedValue();

    await setupLayoutFile();
    const [fileName, content] = writeSpy.mock.lastCall || [];

    expect(fileName).toMatch('src/routes/+layout.svelte');
    expect(content).toMatchSnapshot();
  });

  it('will not setup layout file if it already exists', async () => {
    vi.spyOn(fs, 'existsSync').mockReturnValue(true);
    vi.spyOn(fs, 'ensureFile').mockResolvedValue();
    const writeSpy = vi.spyOn(fs, 'writeFile').mockResolvedValue();

    await setupLayoutFile();

    expect(writeSpy).not.toHaveBeenCalled();
  });

  it('can move postcss.config.js if exists', async () => {
    vi.spyOn(fs, 'existsSync')
      .mockReturnValueOnce(true)
      .mockReturnValueOnce(false);
    const move = vi.spyOn(fs, 'move').mockResolvedValue();

    await movePostCSS();

    const [from, to] = move.mock.lastCall || [];
    expect(move).toHaveBeenCalled();
    expect(from).toMatch('postcss.config.js');
    expect(to).toMatch('postcss.config.cjs');

    await movePostCSS();
    expect(move).toHaveBeenCalledTimes(1);
  });
});
