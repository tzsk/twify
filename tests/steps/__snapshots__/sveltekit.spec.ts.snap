// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SvelteKit Steps > can setup layout file if it does not exist in js 1`] = `
"<script>
  import \\"../app.css\\";
</script>

<slot />
"
`;

exports[`SvelteKit Steps > can setup layout file if it does not exist in ts 1`] = `
"<script lang=\\"ts\\">
  import \\"../app.css\\";
</script>

<slot />
"
`;
